### ISSUE TITLE
PlatformException(4, MEDIA_ELEMENT_ERROR: Format error, null, null)

### ISSUE THREAD

**Issue by saintjab** [bluefireteam/audioplayers#1643]
### Checklist

- [X] I read the [troubleshooting guide](https://github.com/bluefireteam/audioplayers/blob/main/troubleshooting.md) before raising this issue
- [X] I made sure that the issue I am raising doesn't already exist

### Current bug behaviour

Random 
```
AudioPlayerException(
	UrlSource(url: [removed url]),
	PlatformException(4, MEDIA_ELEMENT_ERROR: Format error, null, null)
```
although different attempts to play same urSource worked fine

### Expected behaviour

Correcly play the audio resource at the url

### Steps to reproduce

1. Execute `flutter run` on the code sample 
2. ... 
3. ... 

### Code sample

<details>
<summary>Code sample</summary>

```dart
void main() {
}
```

</details>

### Affected platforms

web

### Platform details

Happened on Safari and Chrome

### AudioPlayers Version

5.1.0

### Build mode

debug, release

### Logs

```
my relevant logs
```

<details>
  <summary>Full Logs</summary>

  

  ```
  my full logs or a link to a gist
  ```

  Flutter doctor:
  ```
  Output of: flutter doctor -v
  ```
</details>

### Related issues / more information

#

### Working on PR

no way

**Comment by Gustl22** [comment]
> [removed url]

Can you actually post a url, that can reproduce the issue?

**Comment by saintjab** [comment]
@Gustl22 no I can't because it random. Issue is when it occurs and I try playing that same url again later, it plays fine. Unfortunately I have moved to using another lib

**Comment by Shubham-VE** [comment]
The audio URLs were working fine before but later I started getting errors. This is the log I get:
```
AudioPlayers Exception: AudioPlayerException(
	UrlSource(url: https://firebasestorage.googleapis.com/v0/b/agzone-teams.appspot.com/o/users%2FXiM6mr1C3ZcLuZVRMHAYVZFVHBf2%2Faudio%2F2023-04-18%2008_27_37.538.mp4?alt=media&token=d670f8d6-cfde-4a73-bd2f-398b44b6103e), 
	PlatformException(WebAudioError, Failed to set source. For troubleshooting, see https://github.com/bluefireteam/audioplayers/blob/main/troubleshooting.md, MediaError: MEDIA_ELEMENT_ERROR: Format error (Code: 4), null)
[2023-10-26T11:24:27.310Z]  @firebase/firestore:
Error: PlatformException(WebAudioError, Failed to set source. For troubleshooting, see https://github.com/bluefireteam/audioplayers/blob/main/troubleshooting.md, MediaError: MEDIA_ELEMENT_ERROR: Format error (Code: 4), null)


dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 327:10  createErrorWithStack
dart-sdk/lib/_internal/js_dev_runtime/patch/core_patch.dart 337:28            _throw
dart-sdk/lib/core/errors.dart 120:5                                           throwWithStackTrace
dart-sdk/lib/async/zone.dart 1386:11                                          callback
dart-sdk/lib/async/schedule_microtask.dart 40:11                              _microtaskLoop
dart-sdk/lib/async/schedule_microtask.dart 49:5                               _startMicrotaskLoop
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 181:15           <fn>
```

**Comment by Gustl22** [comment]
@Shubham-VE this is #1669

**Comment by alanguodev** [comment]
I got the same problem. This is not #1669, as 1669 affects Android as well and only happens in 5.2.0 (reverting to 5.1.0 the problem will go away). This issue only affects web and not all urls (but definitely the url we got back from firebase storage by using  await storage.ref(filePath).getDownloadURL(). And this issue has been there for a long time (I also tried 3.10.10). I am using chrome on Mac.
